src/audio/pipeline/pipeline-graph.c

```diff
@@ -331,13 +331,15 @@ static int pipeline_comp_reset(struct comp_dev *current,
 	pipe_dbg(p_current, "pipeline_comp_reset(), current->comp.id = %u, dir = %u",
 		 dev_comp_id(current), dir);
 
+#if CONFIG_IPC_MAJOR_4
```

Comment:
Ouch, this is pretty bad. :( 

How about if we'd conditionally define IPC4_MOD_ID in src/include/ipc4/module.h so that it would return 0 when building for IPC3. Then we wouldn't need the #ifdefs in pipeline code...?

Reply:
I agree, is not pretty.
I saw we have a lot `#if CONFIG_IPC_MAJOR_4` and `#if CONFIG_IPC_MAJOR_3` in code and I thought we'll stick with this until we'll all use IPC4.
But your proposal is better, I'll update the PR.
Thanks.

Reply:
@iuliana-prodan Actually with "pretty bad" I was more thinking about the regression :) But yes, even better if we can minimize the ifdefs in code. We do have some of them in code, but we should strive to keep the count down.

Reply:
The commit message doesn't seem to be quite correct - commit 45ca3d430 is only touching IPC4 files, so it cannot change IPC3 behaviour as the commit message seems to suggest. Could you clarify?

Reply:
> Ouch, this is pretty bad. :(
> 
> How about if we'd conditionally define IPC4_MOD_ID in src/include/ipc4/module.h so that it would return 0 when building for IPC3. Then we wouldn't need the #ifdefs in pipeline code...?

@kv2019i wait, are you saying, that that file is included in IPC3 builds??

Reply:
btw, we also can use `IS_ENABLED(CONFIG_IPC_MAJOR_4)` inside `if`s instead of using `IPC4_MOD_ID()` as an IPC version check

Reply:
I've added a new version using `IS_ENABLED(CONFIG_IPC_MAJOR_4)` when defining` IPC4_MOD_ID()`

Reply:
@lyakh yes, pipeline-stream.c includes ipc4/module.h unconditionally. otherwise this code would not work at all.

But you raise a good point that "if (IPC4_MOD_ID(current->ipc_config.id))" check is only used to check for IPC4 builds, so we could clean this up further by doing:

```
-»       if (!is_single_ppl && (!is_same_sched || IPC4_MOD_ID(current->ipc_config.id))) {
+»       if (!is_single_ppl && (!is_same_sched || IS_ENABLED(CONFIG_IPC_MAJOR_4))) {

i.e. in IPC4 builds, IPC4_MOD_ID() check is always true. I thought we had some case this would not be the case, but alas it seems really not. Not sure why @lyakh and @RanderWang you did not use this originally. Relying on IPC4_MOD_ID() to be zero in IPC3 builds is really not very readable. So I think this is one case where ifdef is better. 

So @iuliana-prodan if you can do one more update?

Reply:
@kv2019i This
```
-»       if (!is_single_ppl && (!is_same_sched || IPC4_MOD_ID(current->ipc_config.id))) {
+»       if (!is_single_ppl && (!is_same_sched || IS_ENABLED(CONFIG_IPC_MAJOR_4))) {
```
is redundant with the way I define IPC4_MOD_ID now:
`#define IPC4_MOD_ID(x) (IS_ENABLED(CONFIG_IPC_MAJOR_4) ? ((x) & 0xffff) : 0)`